include/audio_module/audio_module.h

```diff
@@ -0,0 +1,500 @@
+ */
+int audio_module_configuration_get(struct audio_module_handle const *const handle,
+				   struct audio_module_configuration *configuration);
+
+/**
+ * @brief Connect two audio modules together.
+ *
+ * @param handle_from[in/out]   The handle for the module for output.
+ * @param handle_to[in/out]     The handle of the module for input, should be NULL if
+ *                              connect_external flag is true.
+ * @param connect_external[in]  Flag to indicate if the from module should put it's output audio
+ *                              data items onto it's TX FIFO for access via an external system.
+ *
+ * @return 0 if successful, error otherwise.
+ */
+int audio_module_connect(struct audio_module_handle *handle_from,
+			 struct audio_module_handle *handle_to, bool connect_external);
+
+/**
+ * @brief Disconnect audio modules from each other.
+ *
+ * @param handle[in/out]             The handle for the module.
+ * @param handle_disconnect[in/out]  The handle of the module to disconnect, should be NULL if
+ *                                   disconnect_external flag is true.
+ * @param disconnect_external[in]    Flag to indicate that the output audio data items should stop
+ *                                   being put on handle TX FIFO.
+ *
+ * @return 0 if successful, error otherwise.
+ */
+int audio_module_disconnect(struct audio_module_handle *handle,
+			    struct audio_module_handle *handle_disconnect,
+			    bool disconnect_external);
+
+/**
+ * @brief Start processing audio data in the audio module given by handle.
+ *
+ * @param handle[in/out]  The handle for the module to start.
+ *
+ * @return 0 if successful, error otherwise.
+ */
+int audio_module_start(struct audio_module_handle *handle);
+
+/**
+ * @brief Stop processing audio data in the audio module given by handle.
+ *
+ * @param handle[in/out]  The handle for the module to be stopped.
+ *
+ * @return 0 if successful, error otherwise.
+ */
+int audio_module_stop(struct audio_module_handle *handle);
+
+/**
+ * @brief Send an audio data item to an audio module, all data is consumed by the module.
+ *
+ * @param handle[in/out]   The handle for the receiving module instance.
+ * @param audio_data[in]   Pointer to the audio data to send to the module.
+ * @param response_cb[in]  Pointer to a callback to run when the buffer is
+ *                         fully consumed in a module.
+ *
+ * @return 0 if successful, error otherwise.
+ */
+int audio_module_data_tx(struct audio_module_handle *handle,
+			 struct audio_data const *const audio_data,
+			 audio_module_response_cb response_cb);
+
+/**
+ * @brief Retrieve an audio data item from an audio module.
+ *
+ * @param handle[in/out]   The handle to the module instance.
+ * @param audio_data[out]  Pointer to the audio data from the module.
+ * @param timeout[in]      Non-negative waiting period to wait for operation to complete
+ *	                       (in milliseconds). Use K_NO_WAIT to return without waiting,
```

Comment:
I suggest the formatting to be:
![image](https://github.com/gWacey/sdk-nrf/assets/89020256/2c967d17-f07e-4d6e-ad74-65541a1db5d8)

Reply:
There were tabs rather than spaces!!!